### PR TITLE
Redesign module positions page

### LINKED ISSUE #21092: [IMPROVEMENT][HOOKS] Redesign module positions page
Hi everyone!

Thank you @PululuK for your report. 

I understand your needs, and your improvement request seems a really nice idea to me. Overall, this hooks page needs a complete redesign since it has exclusive features such as managing hooks using the component that appears on the right side of this page. 

image

This page uses tables to display available hooks but doesn't look like other tables in the bo, see the checkbox on the right side with the number of modules used. In fact, we definitely need to harmonize our components to keep consistency.

**Here is my design proposition:** 

- Bringing back the bulk actions on the left side for each table 
- Being able to activate/deactivate module using the switch
- Using an explicit icon 'remove' for unhook action
- Adding tooltips on each possible actions
- Reduce opacity to each deactivated modules
- Separate each table so it can be easily identifiable

**Screenshots:**
image
image

_Originally posted by @prestascott in https://github.com/PrestaShop/PrestaShop/issues/20848#issuecomment-694100030_

### PR DESCRIPTION
**Pull request by PululuK**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Redesign module positions page.
| Type?             | improvement 
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/21092
| How to test?      | Please see https://github.com/PrestaShop/PrestaShop/issues/21092
| Possible impacts? | N/A

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22595)

### CODE REVIEW

File: src/PrestaShopBundle/Resources/views/Admin/Improve/Design/positions.html.twig [hunk 1]
```diff
@@ -26,66 +26,96 @@
...
+                    <div class="dropdown-divider"></div>
+                    <button class="dropdown-item btn btn-default" name="form_action" value="enable" form="module-positions-form">
+                        <i class="material-icons">radio_button_checked</i>{{ 'Activate selection' | trans({}, 'Admin.Actions') }}
+                    </button>
+                    <button class="dropdown-item btn btn-default" name="form_action" value="disable" form="module-positions-form">
+                        <i class="material-icons">radio_button_unchecked</i>{{ 'Deactivate selection' | trans({}, 'Admin.Actions') }}
+                    </button>
+                    <div class="dropdown-divider"></div>
+                    <button class="dropdown-item btn btn-default" name="form_action" value="unhook" form="module-positions-form">
+                        <i class="material-icons">remove_circle</i>{{ 'Unkook selection' | trans({}, 'Admin.Actions') }}
```
Comment (Julievrz) [inline src/PrestaShopBundle/Resources/views/Admin/Improve/Design/positions.html.twig:108]:
```suggestion
                        <i class="material-icons">remove_circle</i>{{ 'Unhook selection' | trans({}, 'Admin.Actions') }}
```

File: install-dev/upgrade/sql/1.7.8.0.sql [hunk 2]
```diff
@@ -45,3 +45,5 @@ UPDATE `PREFIX_product` SET `location` = '' WHERE `location` IS NULL;
 ALTER TABLE `PREFIX_product` MODIFY COLUMN `location` VARCHAR(255) NOT NULL DEFAULT '';
 UPDATE `PREFIX_product_attribute` SET `location` = '' WHERE `location` IS NULL;
 ALTER TABLE `PREFIX_product_attribute` MODIFY COLUMN `location` VARCHAR(255) NOT NULL DEFAULT '';
+
+ALTER TABLE `PREFIX_hook_module` ADD `active` TINYINT(1) NOT NULL AFTER `position`;
```
Comment (Progi1984) [inline install-dev/upgrade/sql/1.7.8.0.sql:49]: May be it lacks a default value ?

File: src/PrestaShopBundle/Controller/Admin/Improve/Design/PositionsController.php [hunk 3]
```diff
@@ -271,6 +277,130 @@ public function toggleStatusAction(Request $request)
...
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function toggleHookModuleStatusAction(Request $request)
+    {
+        $parameterBag = $request->request;
+        $hookId = (int) $parameterBag->get('hookId');
+        $moduleId = (int) $parameterBag->get('moduleId');
+        $hookStatus = false;
```
Comment (Progi1984) [inline src/PrestaShopBundle/Controller/Admin/Improve/Design/PositionsController.php:294]:
```suggestion
```

File: src/PrestaShopBundle/Controller/Admin/Improve/Design/PositionsController.php [hunk 4]
```diff
@@ -271,6 +277,130 @@ public function toggleStatusAction(Request $request)
...
+    /**
+     * Form actions dispact.
+     *
+     * @AdminSecurity("is_granted(['delete'], request.get('_legacy_controller')~'_')", message="Access denied.")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function formAction(Request $request)
```
Comment (Progi1984) [inline src/PrestaShopBundle/Controller/Admin/Improve/Design/PositionsController.php:377]:
```suggestion
    public function formAction(Request $request): Response
```

File: src/PrestaShopBundle/Resources/views/Admin/Improve/Design/positions.html.twig [hunk 5]
```diff
@@ -26,66 +26,96 @@
 {% trans_default_domain "Admin.Design.Feature" %}
 
 {% block content %}
+<div class='container'>
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Improve/Design/positions.html.twig:29]:
```suggestion
<div class="container"'>
```

File: src/PrestaShopBundle/Resources/views/Admin/Improve/Design/positions.html.twig [hunk 6]
```diff
@@ -26,66 +26,96 @@
...
+                   <option value="all">{{ 'All modules' | trans }}&nbsp;</option>
+                   {% for module in modules %}
+                       <option value="{{ module.id }}"{% if selectedModule == module.id %} selected="selected"{% endif %}>{{ module.displayName }}</option>
+                   {% endfor %}
+               </select>
+             </div>
+             <div class="col-md-8">
+                <h3>{{ 'Search for a hook' | trans }}</h3>
+                <div class="input-group">
+                  <input type="text" class="form-control" id="hook-search" name="hook-search" placeholder="">
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Improve/Design/positions.html.twig:52]:
```suggestion
                  <input type="text" class="form-control" id="hook-search" name="hook-search" />
```

File: src/PrestaShopBundle/Resources/views/Admin/Improve/Design/positions.html.twig [hunk 7]
```diff
@@ -26,66 +26,96 @@
...
-                </div>
+          <div class="mt-3">
+            <div class="row">
+              <div class="col-md-12">
+                <p class="checkbox">
+                  <label class="form-control-label" for="hook-position">
+                  <input type="checkbox" id="hook-position" />
+                  <label class="selectbox" for="hook-position"><i class="material-icons done">done</i></label>
+                    {{ 'Display non-positionable hooks' | trans }}
+                  </label>
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Improve/Design/positions.html.twig:69]:
```suggestion
                  <label class="form-control-label" for="hook-position">
                    <input type="checkbox" id="hook-position" />
                    <label class="selectbox" for="hook-position"><i class="material-icons done">done</i></label>
                    {{ 'Display non-positionable hooks' | trans }}
                  </label>
```

File: src/PrestaShopBundle/Resources/views/Admin/Improve/Design/positions.html.twig [hunk 8]
```diff
@@ -26,66 +26,96 @@
...
+                        <i class="material-icons">remove_circle</i>{{ 'Unkook selection' | trans({}, 'Admin.Actions') }}
+                    </button>
+                </div>
+            </div>
+          {% endif %}
           {% for hook in hooks %}
             <section class="hook-panel{% if not hook['position'] %} hook-position{% endif %}"{% if not hook['position'] %} style="display:none;"{% endif %}>
+            <div class="card">
+                <div class="card-body">
               <a name="{{ hook['name'] }}"></a>
```
Comment (Progi1984) [inline src/PrestaShopBundle/Resources/views/Admin/Improve/Design/positions.html.twig:117]: Please change indentation

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

**Comment by PierreRambaud** [comment]
Hi @PululuK ,

Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**